src/Processors/QueryPlan/Optimizations/convertAnyJoinToSemiOrAntiJoin.cpp

```diff
@@ -217,9 +216,8 @@ size_t tryConvertAnyJoinToSemiOrAntiJoin(QueryPlan::Node * parent_node, QueryPla
                     /// Remove filter after ANTI JOIN because it's a constant expression that always evaluates to TRUE for not matched rows
                     LOG_DEBUG(getLogger("QueryPlanConvertAnyJoinToSemiOrAntiJoin"), "Removing filter '{}' after ANTI JOIN because it's always TRUE for not matched rows", filter_column_name);
                     parent_node->step = convertToExpressionStep(filter);
-                    return 2;
                 }
-                return 1;
+                return 2;
```

Comment:
I can't understand why `2` should be returned here. The plan looks like:
```
Filter -> JoinStepLogical
```

In such a case, `FilterStep` is not modified, and `JoinStepLogical` changes the JOIN operation strictness. This means that only one node was modified.